python/pyspark/ml/util.py

```diff
@@ -283,3 +333,204 @@ def numFeatures(self):
         Returns the number of features the model was trained on. If unknown, returns -1
         """
         return self._call_java("numFeatures")
+
+
+@inherit_doc
+class DefaultParamsWritable(MLWritable):
+    """
+    .. note:: DeveloperApi
+
+    Helper trait for making simple `Params` types writable.  If a `Params` class stores
+    all data as [[pyspark.ml.param.Param]] values, then extending this trait will provide
```

Comment:
There are still some instances of Scaladoc-style class references using double brackets like this: ```[[MyClass]]```.  Switch to the python style like ```:py:class:`MyClass` ```